docs/src/main/asciidoc/cdi-reference.adoc

```diff
@@ -408,6 +426,33 @@ public class CustomTracerConfiguration {
 `@DefaultBean` allows extensions (or any other code for that matter) to provide defaults while backing off if beans of that type are supplied in any
 way Quarkus supports.
 
+=== Declaring Selected Alternatives
+
+In CDI, an alternative bean may be selected either globally for an application by means of `@Priority`, or for a bean archive using a `beans.xml` descriptor.
+Quarkus has a simplified bean discovery and the content of `beans.xml` is ignored.
+
+The disadvantage of `@Priority` is that it has `@Target({ TYPE, PARAMETER })` and so it cannot be used for producer methods and fields.
+To address this problem and to simplify the code Quarkus provides the `io.quarkus.arc.AlternativePriority` annotation.
+It's basically a shortcut for `@Alternative` plus `@Priority`.
+Additionaly, it can be used for producers.
+
+However, it is also possible to select alternatives for an application using the unified configuration.
+The `quarkus.arc.selected-alternatives` property accepts a list of string values.
```

Comment:
How does this work for producer methods? You talk about them in the paragraph above but this section only mentions classes.

Reply:
You should state the declaring class for producers - it's the same as in `beans.xml`. I'll update the docs.

Reply:
Is it better now?